src/wallet/coinselection.cpp

```diff
@@ -155,9 +155,9 @@ bool SelectCoinsBnB(std::vector<CInputCoin>& utxo_pool, const CAmount& target_va
     // Set output set
     value_ret = 0;
     for (size_t i = 0; i < best_selection.size(); ++i) {
```

Comment:
This should go up to a tracked `best_selection_size`. Otherwise we may accidentally include outputs which are not part of the best selection as they are at locations past the size of the actual best selection.